File: CustomerCRUD.API/src/CustomerCRUD/CustomerCRUD.Service/CustomerServices/Implementations/CustomerService.cs
﻿using CustomerCRUD.Core.DTOs;
using CustomerCRUD.Core.Models;
using CustomerCRUD.Service.CustomerServices.Interfaces;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Xml.Linq;

namespace CustomerCRUD.Service.CustomerServices.Implementations
{
    public class CustomerService : ICustomerService
    {
        private readonly List<Customer> _customers;
        public CustomerService()
        {
            _customers = new List<Customer>();
        }

        public List<Customer> Customers => _customers;

        public Customer AddCustomer(CustomerDTO customerDto)
        {
            Customer customer = new Customer();
            customer.Age= customerDto.Age;
            customer.Name= customerDto.Name;
            customer.PostCode= customerDto.PostCode;
            customer.Height= customerDto.Height;
            _customers.Add(customer);
            return customer;
        }

        public Customer UpdateCustomer(Customer customer)
        {
           Customer existingCustomer = _customers.FirstOrDefault(c=>c.Id == customer.Id);
            if (existingCustomer != null)
            {
                existingCustomer.Age= customer.Age;
                existingCustomer.Name= customer.Name;
                existingCustomer.Height= customer.Height;
                return customer;
            }
            else 
            { 
                throw new ArgumentException($"Customer with Id {customer.Id} not found."); 
            }
        }
    }
}

File: CustomerCRUD.UI/src/CustomerCRUD.UI/src/app/services/customer.service.ts
import { Injectable } from '@angular/core';
import { Customer } from '../models/Customer';

@Injectable({
  providedIn: 'root'
})
export class CustomerService {

  constructor() { }

  public getCustomers(): Customer[] {
    let customer = new Customer();
    customer.id="674d3cb7-375f-46b0-8768-0611e32b07b5";
    customer.age=40;
    customer.height=2.00;
    customer.postCode="ABD23";
    customer.name="Gella";

    return [customer];
  }
}
